### PR TITLE
Bug 2098299: Switch to perform normal marshalling with unknown fields

### PR DESCRIPTION
**Pull request by rna-afk**
The hive team uses a lot of fields in the install config that might
not exist due to older versions of the installer missing a few fields
that may have been added in the newer versions, which causes the
installer to reject them all but is needed for proper functioning.

Changing the code a little to now provide a warning message to the
user in case of any unknown fields and then try to unmarshal it
normally instead of strict.

### CODE REVIEW

**Comment by 2uasimojo** [comment]
lgtm, thanks @rna-afk!

File: pkg/asset/installconfig/installconfig.go [hunk 1]
```diff
@@ -131,11 +132,17 @@ func (a *InstallConfig) Load(f asset.FileFetcher) (found bool, err error) {
...
-			err = errors.Wrapf(err, "failed to parse first occurence of unknown field")
-		}
 		err = errors.Wrapf(err, "failed to unmarshal %s", installConfigFilename)
-		return false, errors.Wrap(err, asset.InstallConfigError)
+		if !strings.Contains(err.Error(), "unknown field") {
+			return false, errors.Wrap(err, asset.InstallConfigError)
+		}
+		err = errors.Wrapf(err, "failed to parse first occurence of unknown field")
+		logrus.Warnf(err.Error())
+		logrus.Warnf("trying unmarshal without unknown fields check which may cause errors")
```
Comment (patrickdillon) [inline pkg/asset/installconfig/installconfig.go:141]: 1. Log messages--unlike error messages (which makes this hard to remember)--should be capitalized
2. Let's switch this to an Info
3. Maybe reword to something like
```suggestion
		logrus.Info("Attempting to unmarshal while ignoring unknown keys because strict unmarshaling failed")
```

File: pkg/asset/installconfig/installconfig.go [hunk 2]
```diff
@@ -131,11 +132,17 @@ func (a *InstallConfig) Load(f asset.FileFetcher) (found bool, err error) {
...
+		if err = yaml.UnmarshalStrict(file.Data, config); err != nil {
```
Comment (patrickdillon) [inline pkg/asset/installconfig/installconfig.go:142]:
```suggestion
		if err = yaml.Unmarshal(file.Data, config); err != nil {
```

**Comment by patrickdillon** [comment]
/hold
for tests

**Comment by rna-afk** [comment]
Turns out its due to rate limiting. Closing this then.
https://coreos.slack.com/archives/CBN38N3MW/p1656078735424419

**Comment by rna-afk** [comment]
Errors in testing don't seem to be due to these changes. Removing hold.
/hold cancel

**Comment by patrickdillon** [comment]
> This needs to be backported to 4.10. (Did this PR end up in 4.12? If so, it needs to be backported to 4.11 as well / first.)

@2uasimojo  Can this go into a 4.11.z stream or does it need to go into GA? If it needs to go into GA we need to set this as blocker+ and get staff engineer approval. I think doing that should be pretty clear cut with this change, if we need it.
